xtask/src/tasks.rs

```diff
@@ -21,6 +21,26 @@ pub fn check_formatting(sh: &Shell) -> anyhow::Result<()> {
     Ok(())
 }
 
+pub fn check_typos(sh: &Shell) -> anyhow::Result<()> {
+    let _s = Section::new("TYPOS");
+
+    if cmd!(sh, "typos --version").output().is_err() {
+        println!("`typos-cli` utility is missing, installing...");
+
+        let output = cmd!(sh, "{CARGO} install typos-cli").ignore_status().output()?;
```

Comment:
Since this will run in CI, I prefer we pin a specific version of typos here. Otherwise, we will get different version without intervention as they are published.